lib/index.js

```diff
@@ -15,6 +15,28 @@ function wrapError(callback) {
   };
 }
 
+function putUser(db, user, opts, callback) {
+  var reservedWords = ['name', 'password', 'roles', 'type', 'salt', 'metadata'];
+  if (opts.metadata) {
+    for (var key in opts.metadata) {
+      if (opts.hasOwnProperty(key)) {
+        if (reservedWords.indexOf(key) !== -1 || key.startsWith('_')) {
```

Comment:
This is obvious to couchdb pro's but being a noob - it would be helpful to have 2 error messages here...

Possibly

```
if (reservedWords.indexOf(key) !== -1) {
  return callback(new AuthError('Cannot use reserved word in metadata: "' + key + '"'));
}
if (key.startsWith('_')) {
  return callback(new AuthError('Cannot have metadata key start with an _ as seen on key: "' + key + '"'));
}
```

Thoughts?